interp/notation_term.mli

```diff
@@ -118,7 +126,14 @@ type reversibility_status =
   | Forgetful of forgetfulness
   | NonInjective of Id.t list
 
+type notation_interp_var_types = notation_var_internalization_type Names.Id.Map.t
+
+type 'a notation_raw_type =
+  | NtnRawTypeVar of 'a
+  | NtnRawTypeVarList of ('a * 'a) notation_raw_type (* A .. A *)
+  | NtnRawTypeVarTuple of 'a notation_raw_type list (* A1 * ... * An *)
+
 type notation_interp_env = {
-  ninterp_var_type : notation_var_internalization_type Id.Map.t;
-  ninterp_rec_vars : Id.t Id.Map.t;
+  ninterp_var_type : notation_interp_var_types;
+  ninterp_raw_types : Id.t notation_raw_type list;
 }
```

Comment:
We should probably take the opportunity to document this. For instance, why do we have two ids for list.